homeassistant/components/wallbox/services.yaml

```diff
@@ -0,0 +1,39 @@
+get_schedules:
+  fields:
+    device_id:
+      required: true
+      selector:
+        device:
+          integration: wallbox
+
+set_schedules:
+  fields:
+    device_id:
+      required: true
+      selector:
+        device:
+          integration: wallbox
+    schedules:
+      required: true
+      selector:
+        object:
+      example: >-
+        [
+          {
+            'id': 0,
+            'enable': True,
+            'start': "21:00",
+            'stop': "22:00",
+            'days': {
+              'monday': True,
+              'tuesday': True,
+              'wednesday': True,
+              'thursday': True,
+              'friday': True,
+              'saturday': True,
+              'sunday': True
+            },
+            'max_current': 1,
+            'max_energy': 0
+          }
+        ]
```

Comment:
We should not make users create their own json to put to the service. I'd rather see something smarter that we can validate better. I think there is a risk here since you are editing the full schedule and not just a part of it, so theoretically someone could remove their whole schedule with a single service call (and maybe that isn't a bad feature, but in the current form its too easy for someone to mess up their settings).

Reply:
The original idea was to pass the data easily from get_schedules to set_schedules for this exact reason: the user could get the schedules, modify one and pass it easily to the set_data. I'll look into this and at least create models of the input data that we could validate.

And what comes to easily over write the whole schedules, the Wallbox client lib and the API only supports set functionality at the moment. I think we should not create single schedule handling inside the Home Assistant component.. on this PR at least.

Reply:
@joostlek was just looking into this and could not find a way to define an array to be set on services.yaml.
Is there such functionality in the core that we can 'add a new row' on services.yaml that will create a new set of inputs? I guess 'multiple: true' will only allow to be set in one input field?

Reply:
That's not possible

I think we should try to think of the use cases someone can have with the schedule. Like you said, you introduced both services to do a quick, get data -> add object -> set data, but that could also be just a service that says "add schedule entry" with "from time", "to time", "days", and when you execute it does the same thing, fetch the current schedule, adds the JSON and sets it. Same functionality, but much more user friendly and less easy to mess up.

Reply:
I agree, however, I do require this functionality myself to override the whole schedule from Home Assistant :-)

I'd propose to get forward with this PR using only the get_schedules and set_schedules functionality and I'll start the implementation of set_schedule (single) on another branch + PR. It requires a bit more work since we need to do the get+set requests on the same service call.

Reply:
I'll have to discuss this with some others to get some more opinions. IMO this is too technical and has a too specific use case if it can't be split into separate services

Reply:
Started to look into single schedule set and I think I can do it also on this PR.
However, I would also like to keep the set_schedules if acceptable by HA team :-)

How about removal of a single schedule? Own service call or should we delete a single schedule by providing just an 'id' and empty content? Any suggestions?

Reply:
I think that if we go down the calendar entity route, we don't need the seperate services, right? So why add them now only to remove them later (duplicate functionality) and create a breaking change then? I vote against adding the services now as an interim solution, lets see if a calendar entity can do the same. Also, I don't believe that we should add functionality because 1 user (in this case the developer...) needs it.